recurly/base_client.py

```diff
@@ -25,9 +25,10 @@ def request_converter(value):
 
 
 class BaseClient:
-    def __init__(self, api_key):
+    def __init__(self, api_key, timeout=None):
         self.__api_key = api_key
-        self.__conn = http.client.HTTPSConnection(HOST, PORT)
+        self.timeout = timeout if timeout is not None else DEFAULT_REQUEST_TIMEOUT
+        self.__conn = http.client.HTTPSConnection(HOST, PORT, timeout=self.timeout)
```

Comment:
In the event that everything fails on line `30` and `self.timeout == None`, does the HTTPSConnection class set the timeout to the global (internal) default? the python docs aren't super clear about it.

Reply:
`timeout=None` is the global default (requests wait forever) so it would essentially be a no-op